runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -6235,6 +6235,19 @@ static int32_t J9THREAD_PROC samplerThreadProc(void * entryarg)
             // compute jit state
             jitStateLogic(jitConfig, compInfo, diffTime); // Update JIT state before going to sleep
 
+            // detect high code cache occupancy and increase initial invocation counts
+            size_t usedCacheSize = TR::CodeCacheManager::instance()->getCurrTotalUsedInBytes() / 1000;
+            UDATA totalCacheSize = jitConfig->codeCacheTotalKB;
+            if ((usedCacheSize / 3) >= (totalCacheSize / 4))
```

Comment:
It's better to have an option that determines when to increase the counts

Reply:
We can avoid divisions with a static variable:
Occupancy (as a percentage) is calculated as usedSize/totalSize*100. The `if` statement is
`usedSize/totalSize*100 > TR::Options::getHighCodeCacheOccupancyThreshold`
which can be rewritten as
`usedSize >  TR::Options::getHighCodeCacheOccupancyThreshold * (totalSize /100)`
and `TR::Options::getHighCodeCacheOccupancyThreshold * totalSize /100` can be memorized in a static variable (making sure to avoid overflow)